frontend/src/metabase/visualizations/shared/settings/cartesian-chart.ts

```diff
@@ -327,8 +327,10 @@ export const isXAxisScaleValid = (
   }
 
   return (
-    !isWaterfall ||
-    (xAxisScale && !WATERFALL_UNSUPPORTED_X_AXIS_SCALES.includes(xAxisScale))
+    (!isWaterfall ||
+      (xAxisScale &&
+        !WATERFALL_UNSUPPORTED_X_AXIS_SCALES.includes(xAxisScale))) ??
+    false
```

Comment:
Won't `(!isWaterfall ||
      (xAxisScale &&
        !WATERFALL_UNSUPPORTED_X_AXIS_SCALES.includes(xAxisScale)))` always evaluate to Boolean? Not sure why we need the `?? false` coalescing

Reply:
So _technically_ `xAxisScale` can evaluate to `null | undefined` (on a different branch rn so I can't check exactly), so the function would return `boolean | (null | undefined)`. The function using this needs to return `boolean` only so I had to coerce it. We could also coerce it there but considering this is a boolean function I would rather do it here

Reply:
Ooh, I forgot that `null && false` evaluates to `null` and `undefined && false` evaluates to `undefined`. JavaScript biting me in the ass again

Reply:
I'd just wrap the whole thing in `Boolean(xxxxx)` rather than `?? false at the end. What do you think?

Reply:
works for me